plugins/utils/refactoring.nix

```diff
@@ -0,0 +1,165 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+with lib; let
+  helpers = import ../helpers.nix {inherit lib;};
+in {
+  options.plugins.refactoring =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "refactoring.nvim";
+
+      package = helpers.mkPackageOption "refactoring.nvim" pkgs.vimPlugins.refactoring-nvim;
+
+      promptFuncReturnType =
+        helpers.defaultNullOpts.mkNullable
+        (
+          with types;
+            attrsOf bool
+        )
+        ''
+          {
```

Comment:
Im not sure if this it the best approach ?

Reply:
I think it is